src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -175,23 +174,58 @@ public override void Initialize(AnalysisContext context)
 
                 context.RegisterOperationBlockStartAction(
                     context => AnalyzeOperationBlock(context, guardMethods, runtimeIsOSPlatformMethod, osPlatformCreateMethod, crossPlatform,
-                                    osPlatformType, stringType, platformSpecificMembers, msBuildPlatforms, notSupportedExceptionType));
+                                    osPlatformType, stringType, platformSpecificMembers, msBuildPlatforms, notSupportedExceptionType, relatedPlatforms));
             });
 
-            static ImmutableArray<IMethodSymbol> GetOperatingSystemGuardMethods(IMethodSymbol? runtimeIsOSPlatformMethod, INamedTypeSymbol operatingSystemType)
+            static ImmutableArray<IMethodSymbol> GetOperatingSystemGuardMethods(IMethodSymbol? runtimeIsOSPlatformMethod,
+                INamedTypeSymbol operatingSystemType, INamedTypeSymbol? mockOSType, out SmallDictionary<string, string> relatedPlatforms)
             {
-                var methods = operatingSystemType.GetMembers().OfType<IMethodSymbol>().Where(m =>
-                    m.IsStatic &&
-                    m.ReturnType.SpecialType == SpecialType.System_Boolean &&
-                    (IsOSPlatform == m.Name) || NameAndParametersValid(m)).
-                    ToImmutableArray();
+                relatedPlatforms = new SmallDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                var methods = FilterPlatformCheckMethods(operatingSystemType, relatedPlatforms).ToList();
+
+                if (mockOSType != null)
+                {
+                    methods.AddRange(FilterPlatformCheckMethods(mockOSType, relatedPlatforms));
+                }
 
                 if (runtimeIsOSPlatformMethod != null)
                 {
-                    return methods.Add(runtimeIsOSPlatformMethod);
+                    methods.Add(runtimeIsOSPlatformMethod);
                 }
 
-                return methods;
+                return methods.ToImmutableArray();
+
+                static IEnumerable<IMethodSymbol> FilterPlatformCheckMethods(INamedTypeSymbol symbol, SmallDictionary<string, string> relatedPlatforms)
+                {
+                    return symbol.GetMembers().OfType<IMethodSymbol>().Where(m =>
+                    {
+                        if (m.IsStatic &&
+                            m.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                            (IsOSPlatform == m.Name) || NameAndParametersValid(m))
+                        {
+                            CheckDependentPlatofmrs(m, relatedPlatforms);
```

Comment:
Typo in method name `CheckDependentPlatofmrs`.

```suggestion
                            CheckDependentPlatforms(m, relatedPlatforms);
```